packages/core/react/src/WalletProvider.tsx

```diff
@@ -112,6 +112,11 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     // Handle the adapter's disconnect event
     const onDisconnect = useCallback(() => setName(null), [setName]);
 
+    const onAccountChanged = useCallback(async () => {
+        await disconnect();
+        await connect();
+    }, []);
```

Comment:
I don't think this will produce a very good UX. Disconnecting and reconnecting the wallet isn't necessary if the change has already occurred (for example, indicating that a new `publicKey` is available).